hls-plugin-api/src/Ide/Types.hs

```diff
@@ -657,6 +660,10 @@ instance PluginRequestMethod Method_CodeLensResolve where
 
 instance PluginRequestMethod Method_TextDocumentRename where
 
+instance PluginRequestMethod Method_TextDocumentPrepareRename where
+    -- TODO more intelligent combining?
+    combineResponses _ _ _ _ (x :| _) = x
```

Comment:
I think it is fine we only pick up the newest

Reply:
I think this is used combining or responses in case multiple plugins happen to handler this type of request.
I'd say it's ok to leave it like this for now, because this PR introduces this new request type and for now rename plugin is the only one that supports this kind of request.